evm/vm_state.py

```diff
@@ -236,6 +241,33 @@ def get_transaction_context_class(cls):
     #
     # Execution
     #
+
+    def do_call(self, transaction):
+
+        _transaction = transaction
+
+        _transaction.v = 37
+        _transaction.s = 1
+        _transaction.r = 1
+
+        snapshot = self.snapshot()
+
+        with self.mutable_state_db() as state_db:
+
+            if not hasattr(_transaction, "get_sender"):
+                # max out eth in random hardcoded address. what could go wrong?
+                _transaction.get_sender = \
+                    lambda: to_bytes(hexstr='0x7760c6D8c0eA7A764F13fb5b034A574f59c1fdd4')
+                _transaction.sender = to_bytes(hexstr='0x7760c6D8c0eA7A764F13fb5b034A574f59c1fdd4')
+
+            state_db.set_balance(transaction.sender, UINT_256_MAX, validate=False)
+
+        computation = self.execute_transaction(_transaction)
+
+        self.revert(snapshot)
```

Comment:
This should probably go in a `finaly` block of a `try/finally` to ensure that the reversion happens even if an exception occurs.